rest/model/project_event.go

```diff
@@ -248,3 +260,15 @@ func (vars *APIProjectVars) IsPrivate(key string) bool {
 	}
 	return utility.StringSliceContains(vars.PrivateVarsList, key)
 }
+
+func (g *APIGithubAppAuth) ToService() *evergreen.GithubAppAuth {
+	return &evergreen.GithubAppAuth{
+		AppID:      int64(g.AppID),
+		PrivateKey: []byte(utility.FromStringPtr(g.PrivateKey)),
+	}
+}
+
+func (g *APIGithubAppAuth) BuildFromService(v evergreen.GithubAppAuth) {
+	g.AppID = int(v.AppID)
+	g.PrivateKey = utility.ToStringPtr(string(v.PrivateKey))
```

Comment:
Should this be set when we're building the API model? Since we never plan to expose it in a user-facing way, it seems more secure to just not set it at all (or remove the field), so it can never be read out from Evergreen.

Reply:
I think the only reason it's done this way is in order to support the event logs, since we want to know if the new `PrivateKey` is different from the old `PrivateKey` 😔 We don't redact the private variables in this file either ([code here](https://github.com/evergreen-ci/evergreen/blob/737621f087c6479325715da01747956d2de0dfe5/rest/model/project_event.go#L159-L163)) and they are saved in the events collection as unredacted. (Which might be bad haha)